src/components/modals/ImmatriculationHistoryModal.vue

```diff
@@ -0,0 +1,60 @@
+<template>
+    <modal ref="baseModal" :action="action" @cancel="close(action.CANCEL)">
+        <template #header>
+            <p class="text-2xl text-gray-900">Immatriculation History</p>
+        </template>
+        <div v-if="busy" class="w-full items-center justify-center">
+            <loading-spinner />
+        </div>
+        <immatriculation-history v-if="alreadyLoadedOnce" v-model:busy="busy" :username="username" />
+        <template #footer>
+            <button id="immatriculationHistoryCancel" class="mr-10 btn-tertiary" @click="close(action.CANCEL)">Cancel</button>
+        </template>
+    </modal>
+</template>
+
+<script lang="ts">
+    import Modal from "@/components/modals/Modal.vue";
+    import { ref, onBeforeMount } from "vue";
+    import ImmatriculationHistory from "@/components/ImmatriculationHistory.vue";
+    import LoadingSpinner from "@/components/loading/Spinner.vue";
+    import { useStore } from "@/store/store";
+
+    export default {
+        components: {
+            Modal,
+            ImmatriculationHistory,
+            LoadingSpinner,
+        },
+        setup() {
+            const baseModal = ref();
+            let busy = ref(false);
+            let alreadyLoadedOnce = ref(false);
```

Comment:
I like caching, but I do not like caches, that are never invalidated.
We should add a "refresh" button to the modal (or somewhere). 

As a student, I would expect the data to refresh automatically, if I open the modal, though.

Reply:
I removed the caching and added a refresh option for the history modal.